app/controllers/user.js

```diff
@@ -18,4 +37,41 @@ userController.addUserWithFacebook = function(request, response) {
     });
 };
 
+userController.sendPasswordResetEmail = function(userEmail) {
+    crypto.randomBytes(20, function(err, buffer) {
+        if (err) {
+            console.log('Failed to generate reset token');
+        } else {
+            var token = buffer.toString('hex');
+            var today = new Date().getTime();
+            var expiration = new Date(today + 86400000);
+            var passwordReset = new PasswordReset({
+                email: userEmail,
+                resetToken: token,
+                expiration: expiration
+            });
+            passwordReset.save()
+            .then(function(resetObject) {
+                var mail = {
```

Comment:
Could potentially even move this into an instance method on the PasswordReset model: `resetObject.sendResetEmail()`